### DISCUSSION TITLE
Is there a more flexible way to specify OTEL_RESOURCE_ATTRIBUTES?

### DISCUSSION THREAD

**Discussion by ncabanis** [open-telemetry/opentelemetry-java-instrumentation#2902] [Q&A]
I am looking for a way to specify the [K8s conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/k8s.md), like `k8s.node.name`, `k8s.namespace.name` or `k8s.pod.name`. As I understood, the only possibility is to specify these fields comma separated within the `OTEL_RESOURCE_ATTRIBUTES` parameter.

Is there not a more flexible way, e.g. by specifying `K8S_NODE_NAME` or maybe `OTEL_RESOURCE_ATTRIBUTES_K8S_NODE_NAME`?

This composite field makes it difficult to provide via an K8s deployment descriptor.

**Comment by mateuszrzeszutek** [comment]
Hey @ncabanis ,

Unfortunately right now there's no way to specify resource attributes in separate env vars.

In general, currently we support the following 3 ways of adding resource attributes:
* the `OTEL_RESOURCE_ATTRIBUTES` environment variable;
* the `otel.resource.attributes` Java system property;
* and a custom `ResourceProvider` implementation - this one requires making a custom javaagent distro though (see https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/2763 for similar inquiry). We're currently working on making this more accessible.
